plugins/connection/podman.py

```diff
@@ -148,10 +153,11 @@ def exec_command(self, cmd, in_data=None, sudoable=False):
 
         # shlex.split has a bug with text strings on Python-2.6 and can only handle text strings on Python-3
         cmd_args_list = shlex.split(to_native(cmd, errors='surrogate_or_strict'))
+        exec_args_list = ["exec"]
         if self.user:
-            cmd_args_list += ["--user", self.user]
+            exec_args_list.extend(("--user", self.user))
```

Comment:
Something is wrong here, from the test:
```
TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
task path: /tmp/test2.yml:2
<podman-container> RUN [b'/usr/bin/podman', b'mount', b'mount', b'podman-container']
Failed to mount container podman-container: b'Error: cannot mount using driver overlay in rootless mode'
<podman-container> RUN [b'/usr/bin/podman', b'exec', b'--user', b'nobody', b"['exec', '--user', 'nobody']", b'podman-container', b'/bin/sh', b'-c', b'echo ~nobody && sleep 0']
<podman-container> RUN [b'/usr/bin/podman', b'exec', b'--user', b'nobody', b"['exec', '--user', 'nobody']", b'podman-container', b'/bin/sh', b'-c', b'echo "`pwd`" && sleep 0']
<podman-container> RUN [b'/usr/bin/podman', b'exec', b'--user', b'nobody', b"['exec', '--user', 'nobody']", b'podman-container', b'/bin/sh', b'-c', b'( umask 77 && mkdir -p "` echo ~/.ansible/tmp `"&& mkdir ~/.a
nsible/tmp/ansible-tmp-1587249105.6848736-22218-208278136013092 && echo ansible-tmp-1587249105.6848736-22218-208278136013092="` echo ~/.ansible/tmp/ansible-tmp-1587249105.6848736-22218-208278136013092 `" ) && sl
eep 0']
fatal: [podman-container]: UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to create temporary directory.In some cases, you may have been able to authenticate and did not have permissions on the target directory. Consider changing the remote tmp path in ansible.cfg t
o a path rooted in \"/tmp\", for more error information use -vvv. Failed command was: ( umask 77 && mkdir -p \"` echo ~/.ansible/tmp `\"&& mkdir ~/.ansible/tmp/ansible-tmp-1587249105.6848736-22218-20827813601309
2 && echo ansible-tmp-1587249105.6848736-22218-208278136013092=\"` echo ~/.ansible/tmp/ansible-tmp-1587249105.6848736-22218-208278136013092 `\" ), exited with result 125, stderr output: Error: no container with 
name or ID ['exec', '--user', 'nobody'] found: no such container\n",
    "unreachable": true
}

```

for reproducing:
```bash
podman run -d --name podman-container python:3-alpine sleep 1d
ansible-playbook -vvvv test2.yml -i podman-container, -u nobody
```
when test2.yml is:
```yaml
---
- hosts: podman-container
  connection: containers.podman.podman
  tasks:

    - name: Print who am I
      shell: whoami
```

**please be aware to use collections connection, not the build one in 2.9**

Reply:
With new patch works like a charm, thanks. Will merge it as we'll have CI working in next few days.